src/backend/distributed/executor/CitusExecutorLockMap.txt

```diff
@@ -0,0 +1,22 @@
+ReadOnly
+	citus.all_modifications_commutative
+		RequiresParallelExecution
+			 TableReplicated
+				NoLock
+				Commutative Modification
+					ExclusiveLock
+					RowExclusiveLock
+			TableReplicated
+				IsCoordinator
+					RowExclusive Lock
+					citus.enable_deadlock_preventio
+						ShareUpdateExclusiveLock
+						RowExclusiveLock
+				ExclusiveLock
+		RowExclusiveLock
+	Select For Update On Reference Table
+		NoLock
+		For Share
+			ExclusiveLock
+			ShareLock
```

Comment:
This does help show when we we take which lock, though the more crucial information is in why we take the lock and what it blocks. 

Maybe we add this to the wiki?